src/filesystem/fs_misc.cpp

```diff
@@ -0,0 +1,965 @@
+/*
+===========================================================================
+Copyright (C) 1999-2005 Id Software, Inc.
+Copyright (C) 2017 Noah Metzger (chomenor@gmail.com)
+
+This file is part of Quake III Arena source code.
+
+Quake III Arena source code is free software; you can redistribute it
+and/or modify it under the terms of the GNU General Public License as
+published by the Free Software Foundation; either version 2 of the License,
+or (at your option) any later version.
+
+Quake III Arena source code is distributed in the hope that it will be
+useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Quake III Arena source code; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+===========================================================================
+*/
+
+#ifdef NEW_FILESYSTEM
+#include "fslocal.h"
+
+/* ******************************************************************************** */
+// Hash Table
+/* ******************************************************************************** */
+
+void fs_hashtable_initialize(fs_hashtable_t *hashtable, int bucket_count)
+{
+    // Valid for an uninitialized hash table
+    hashtable->bucket_count = bucket_count;
+    hashtable->buckets = (fs_hashtable_entry_t **)Z_Malloc(sizeof(fs_hashtable_entry_t *) * bucket_count);
```

Comment:
Z_Malloc and friends are garbage and I'd rather you use `malloc`. If nothing else, because my debug builds all use address sanitizer.

Reply:
To elaborate on this; `Z_Malloc` prevents memory security devices (such as heap cookies), but also prevent tools like address sanitizer and valgrind.